ludwig/data/preprocessing.py

```diff
@@ -1113,7 +1132,8 @@ def build_dataset(
         backend=LOCAL_BACKEND,
         random_seed=default_random_seed,
         skip_save_processed_input=False,
-        callbacks=[]
+        callbacks=[],
+        context=None
```

Comment:
`context` can get overloaded, so maybe something like `mode` will be more natural.

Reply:
Done.